### ISSUE TITLE
Regression when Querying by reference in `$match` stages

### ISSUE THREAD

**Issue by DerNamenlose** [MorphiaOrg/morphia#1709]
**Describe the bug**
When using Query-by-reference in the `$match` stage of an aggregation, no objects are found. The bug seems to be closely related to #1647. The `Filters.eq("field", reference.getID())` pattern works around the issue.

**To Reproduce**
Basically use the example from #1647 in the `$match` stage of an aggregation query.

**Expected behavior**
Query by reference should work as expected in aggregations.

**Please complete the following information:**
 - Server Version: 5.0.2
 - Driver Version: 4.2.2
 - Morphia Version: 2.2.3

**Comment by evanchooly** [comment]
I have a fix in mind for this but it will require [this PR](https://github.com/mongodb/mongo-java-driver/pull/846) for the "proper fix."  If I can't get something like that it will involve some deeper surgery that i'm trying to avoid.  we'll what the response to that PR (hopefully in a few days) before I do too much more work.

**Comment by evanchooly** [comment]
huh.  well, that driver PR probably isn't going to happen for $reasons so I started up on that surgery i was trying to put off.  Turns out, it was 90% done already (it worked for queries) and I just had to account for the aggregation code path.  So I think i have it fixed for you.  I'll tidy up and push it up to github.  Once the actions are done, it should autopublish a `2.2.4-SNAPSHOT` build if you'd like to try it locally before we officially put this one to bed.
